### PR TITLE
Attempt to fix #43

### LINKED ISSUE #43: Pierre allows a PR to self-reference
Should not be able to do that, creates a cyclic reference

### PR DESCRIPTION
**Pull request by rajalakshmi-v15**
Hi. This is my attempt at fixing issue #43 . Thanks.

### CODE REVIEW

File: README.md
Comment (alvarocavalcanti) [inline README.md:13]: Good catch! 👍

File: lib/pierre.py
Comment (alvarocavalcanti) [inline lib/pierre.py]: I would like to have this renamed, since this isn't really the dependency id, it's the pull request id. Something in the likes of `pull_request_id` or `root_id`?

File: lib/pierre.py
Comment (mosheeshel) [inline lib/pierre.py]: you can avoid this new with a simple addition in the `if` above
`if deps && deps != dependency_id:`

**Review by alvarocavalcanti — CHANGES_REQUESTED** [review]

File: README.md
Comment (mosheeshel) [inline README.md:13]: in the future please split different unconnected changes into separate PRs

File: lib/pierre.py
Comment (mosheeshel) [inline lib/pierre.py]: rename `dependency_id` to `this_pr_id` or something that explains that we want this PRs number/id

**Review by mosheeshel — CHANGES_REQUESTED** [review]
@rajalakshmi-v15 looks very good! a few small suggestions

**Comment by alvarocavalcanti** [comment]
@rajalakshmi-v15 Please be aware of the failing CI build. Also, we should have at least a test for this behavior.

File: lib/pierre.py
Comment (mosheeshel) [inline lib/pierre.py]: you can change to 
`return event.get("issue", {}).get("number", "")`
no need to assign and then return

**Comment by mosheeshel** [comment]
@rajalakshmi-v15 please squash your commits into one.

**Comment by rajalakshmi-v15** [comment]
Hi. `./lib/pierre.py:181:1: W293 blank line contains whitespace` is the error the CI build gives me. I went back to line number 181 and removed the extra white spaces. May I know what I should do for that?

File: README.md
Comment (rajalakshmi-v15) [reply README.md:13]: Sorry, @mosheeshel . I thought that the earlier request has already been merged.

**Comment by mosheeshel** [comment]
@rajalakshmi-v15 I'm not sure but I'm assuming the build checks for some standards python style rules (like [PEP8](https://www.python.org/dev/peps/pep-0008/?))

File: README.md
Comment (alvarocavalcanti) [reply README.md:13]: It was not merged, you simply closed the PR. I'm currently the only one that can merge PRs, so a contributor should only check for any outstanding comments and wait for approval. Once it's approved I can merge it.

**Comment by alvarocavalcanti** [comment]
Yes, the CI currently runs `flake8` which checks for PEP8 compliance and other code style. You can run it locally, tho. Just activate the virtual environment `source env/bin/activate`, install the dependencies `pip install -r requirements.txt` and then run `flake8 .`.

**Comment by rajalakshmi-v15** [comment]
The CI build suggests that `test_pierre.py` should also be changed to incorporate the `root_id` argument in `pierre.py`. Can I make changes to 'test_pierre.py`, too?

**Comment by alvarocavalcanti** [comment]
@rajalakshmi-v15 Yes, you can. And also, I would like you to add a new test for this new feature as well.

**Comment by alvarocavalcanti** [comment]
@rajalakshmi-v15 You can run the tests locally using: `py.test tests` That way you can have a faster feedback when trying to fix them.

**Comment by rajalakshmi-v15** [comment]
@alvarocavalcanti, thanks for the tip. Hope I didn't cause much damage.

**Comment by mosheeshel** [comment]
@rajalakshmi-v15 it's no problem TravisCI gives this at no cost, it's just much faster locally

**Comment by rajalakshmi-v15** [comment]
@mosheeshel, Please review the changes. Thanks.

**Review by mosheeshel — APPROVED** [review]
LGTM, please squash into a single commit

**Review by alvarocavalcanti — APPROVED** [review]

**Comment by alvarocavalcanti** [comment]
@rajalakshmi-v15 I have approved the changes, but I'd like to have the commits squashed as @mosheeshel mentioned. After that I'll be happy to merge it. 😁

**Comment by rajalakshmi-v15** [comment]
@mosheeshel, Is it fine now?

**Comment by mosheeshel** [comment]
no :disappointed:  please re-open the PR, you did some wrong rebasing I think - you indeed squashed all your commits into one, but appear to be coming out of the master branch in a wrong point, so a lot of additional commits got added... can you revert it back to the original? @alvarocavalcanti might be able to squash it himself

**Comment by rajalakshmi-v15** [comment]
@mosheeshel, sorry for the trouble. I am not able to reopen the pull request since we don't have a common history. Is there any other way around? Should I open a new issue?

**Comment by mosheeshel** [comment]
I'm guessing you need to pull the latest history from the original repo first and then reopen the PR or open a new one.
this might help https://help.github.com/articles/syncing-a-fork/

**Pull request closed without merging**
